src/templates/filter-bar-controller.js

```diff
@@ -19,7 +18,7 @@ FilterBarController.prototype.selectFilter = function(filterName) {
         return;
     }
 
-    this.selected = Filters[filterName]();
+    store.dispatch(actions.ui.setFilter(filterName));
```

Comment:
if we're going to be saying that "changing the filter means unselecting the note" and we now have a filter change action, we should let this filter action invoke the change in the selected note and remove the call a few lines down to `actions.ui.unselectNote()`, instead handling it in the `selectedNote` reducer